ArduCopter/esc_calibration.cpp

```diff
@@ -212,3 +156,37 @@ void Copter::esc_calibration_notify()
         notify.update();
     }
 }
+
+void Copter::esc_calibration_setup()
+{
+    // clear esc flag for next time
+    g.esc_calibrate.set_and_save(ESCCAL_NONE);
+
+    if (motors->get_pwm_type() >= AP_Motors::PWM_TYPE_ONESHOT) {
+        // run at full speed for oneshot ESCs (actually done on push)
+        motors->set_update_rate(g.rc_speed);
+    } else {
+        // reduce update rate to motors to 50Hz
+        motors->set_update_rate(50);
+    }
+
+    // disable safety if requested
+    BoardConfig.init_safety();
+
+    // wait for safety switch to be pressed
+    uint32_t tstart = AP_HAL::millis();
```

Comment:
Probably should actually default this to 0, so that you have a tendency to print immediately the first time if the switch is in the wrong position.